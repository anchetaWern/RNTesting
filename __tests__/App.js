import "react-native";
import React from "react";
import App from "../App";

import Card from "../src/components/Card";
import pokemonData from "./data/pokemon.json";

import renderer from "react-test-renderer";

it("App is rendered correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("card renders correctly", () => {
  const tree = renderer
    .create(<Card fetching={false} data={pokemonData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
