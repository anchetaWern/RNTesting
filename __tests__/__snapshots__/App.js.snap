// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is rendered correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="action_button"
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/img/pokeball.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "marginTop": 50,
            "width": 120,
          }
        }
        testID="pokeball_image"
      />
    </View>
  </View>
</View>
`;


exports[`card renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
      }
    }
  >
    <View>
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/304.png",
          }
        }
        style={
          Object {
            "height": 120,
            "width": 120,
          }
        }
        testID="pokemon_sprite"
      />
    </View>
    <View
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
        testID="pokemon_name"
      >
        aron
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
      testID="pokemon_types"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#b8a038",
            },
            Object {
              "alignItems": "center",
              "padding": 5,
              "width": 100,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          rock
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#b8b8d0",
            },
            Object {
              "alignItems": "center",
              "padding": 5,
              "width": 100,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          steel
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#3e3e3e",
          }
        }
        testID="pokemon_description"
      >
        Aron has a body of steel. With one all-out charge, this
Pokémon can demolish even a heavy dump truck.
The destroyed dump truck then becomes a handy meal
for the Pokémon.
      </Text>
    </View>
  </View>
</View>
`;
